### ISSUE TITLE
[Feature Request] Make Economy/Industry Supply/Demand based

### ISSUE THREAD

**Issue by Huperspace** [jamespetts/simutrans-extended#112]
If I read the town hall info, I only get 1k demand for ind./com. and -1k for res. (if it's readable at all because of the unscaling). But no playable Industries nor np Ind./Com. are build.
The only thing that spawns industry is the increase_every_<number>. Beside that counts also for expanding(Link to free industry)/upgrading/closing down.
The spawing is only random with %-Changes which is more likely and which not.
I don't want to see two same Industries/Consumers (like 2 markets with 2 sepeate fruit gardens) with sepeate suppliers in the same town. I don't want to see if I have sheep farms and a texilemill that produces more a month than 1 clothes shop can consume a year, and the new industry that spawns is an entire Meat chain that spawns a Slaughterhause with sheep farms at the other side of the maps. I want to see that the new industry would be another clothes shops in near towns up to the mill is fully occupied or a Slaughter House near the allready there sheep farms.

In short at least I want the fully usage of the industy Chains capability before spawning a new separate one.

I don't like the closing down and upgrade event either.
I request if e.g. the invention of Electricity highers the output of the 2 textil mills that supply the entire map with clothes so much that 1 mill is closing down and the sheep farms get linked to that remaining one as well as the clothes shops.
